test/langtools/jdk/javadoc/doclet/checkLibraryVersions/CheckLibraryVersions.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8293177
+ * @summary Verify version numbers in legal files
+ * @run main CheckLibraryVersions
+ */
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Checks the names and version strings of 3rd party libraries in legal files
+ * against the actual names and versions in library files.
+ */
+public class CheckLibraryVersions {
+    static class SourceDirNotFound extends Error {}
+    // Regex pattern for library name and version in legal Markdown file
+    static final Pattern versionPattern = Pattern.compile("## ([\\w\\s]+) v(\\d+\\.\\d+\\.\\d+)");
+
+    // Map of 3rd party libraries. The keys are the names of files in the legal directory,
+    // the values are lists of templates for library files with the following placeholders:
+    //  %V is replaced with the version string
+    //  %M is replaced twice, once with an empty string and once with ".min"
+    static final Map<String, List<String>> libraries = Map.of(
+            "jquery.md", List.of("jquery-%V%M.js"),
+            "jqueryUI.md", List.of("jquery-ui%M.js", "jquery-ui%M.css")
+    );
+
+    public static void main(String... args) throws Exception {
+        try {
+            new CheckLibraryVersions().run(args);
+        } catch (SourceDirNotFound e) {
+            System.err.println("NOTE: Cannot find src directory; test skipped");
```

Comment:
These days, it is common practice (but not required) to throw `jtreg.SkippedException` to convey this info.

Reply:
Right, I changed the code to use `jtreg.SkippedException`.